rust/ommx/src/constraint_hints.rs

```diff
@@ -0,0 +1,62 @@
+use crate::v1::{ConstraintHints, OneHot};
```

Comment:
You don't have to put this into separate file. Generally, you MUST NOT add functions for types in `ommx::v1::*`, as they are generated automatically.

In this case, do the following things:

1. Add `KHot` struct to `rust/ommx/src/instance.rs` and implement `Parse` impl for `v1::KHot` with output set to newly defined `KHot`.
2. Tweak `ConstraintHints` struct in `rust/ommx/src/instance.rs` so that it reflects the raw Constraint Hints.
3. Move `one_hot_constraints` and `get_k_hot_constraints` to impl of `ConstraintHints` in `rust/ommx/src/instance.rs`, changing return type to the ones definedin `rust/ommx/src/instance.rs` , instead of v1 types.